markdown/extensions/md_in_html.py

```diff
@@ -36,9 +36,10 @@ def __init__(self, md, *args, **kwargs):
         # Block-level tags which never get their content parsed.
         self.raw_tags = ['canvas', 'math', 'option', 'pre', 'script', 'style', 'textarea']
         # Block-level tags in which the content gets parsed as blocks
-        self.block_tags = [tag for tag in self.block_level_tags if tag not in self.span_tags + self.raw_tags]
         super().__init__(md, *args, **kwargs)
 
+        self.block_tags = [tag for tag in self.block_level_tags if tag not in self.span_tags + self.raw_tags + self.empty_tags]
```

Comment:
Btw, I see you first tried to use sets here. The appropriate way to do that would be:

```python
self.block_tags = set(self.block_level_tags) - set(self.span_tags + self.raw_tags + self.empty_tags)
```

I debated which way to go with it. In the end I chose to keep using lists rather than changing to sets. I did not check performance. If sets are faster, then by all means, feel free to make the change.